src/common/cli_wrapper.h

```diff
@@ -0,0 +1,275 @@
+
+  std::string keyName(const std::string &args) const {
+    return std::get<1>(CLI::detail::get_names(CLI::detail::split_names(args)))
+        .front();
+  }
+
+public:
+  /**
+   * @brief Create an instance of the command-line parser
+   *
+   * Option --help, -h is automatically added.
+   *
+   * @param name Header for the main option group
+   * @param columnWidth Width of the column with option names
+   * @param screenWidth Maximum allowed width for help messages
+   */
+  CLIWrapper(const std::string &name = "General options",
+             size_t columnWidth = 35,
+             size_t screenWidth = 0);
+
+  virtual ~CLIWrapper();
+
+  /**
+   * @brief Define an option with a default value
+   *
+   * @param args Comma-separated list of short and long option names
+   * @param help Help message
+   * @param val Default value
+   *
+   * @return Option object
+   */
+  template <typename T>
+  CLI::Option *add(const std::string &args, const std::string &help, T val) {
+    return add_option<T>(keyName(args), args, help, val, true, true);
+  }
+
+  /**
+   * @brief Define an option without an explicit default value. The implicit
+   * default value is T()
+   *
+   * The option will be defined in the config file even if not given as a
+   * command-line argument. The implicit default value for a numeric option is
+   * 0, for a string is an empty string, and for a vector is an empty vector.
+   *
+   * @param args Comma-separated list of short and long option names
+   * @param help Help message
+   *
+   * @return Option object
+   */
+  template <typename T>
+  CLI::Option *add(const std::string &args, const std::string &help) {
+    return add_option<T>(keyName(args), args, help, T(), false, true);
+  }
+
+  /**
+   * @brief Define a non-defaulted option
+   *
+   * The option will not be present in the config file unless given as a
+   * command-line argument.
+   *
+   * @param args Comma-separated list of short and long option names
+   * @param help Help message
+   *
+   * @return Option object
+   */
+  template <typename T>
+  CLI::Option *add_nondefault(const std::string &args,
+                              const std::string &help) {
+    return add_option<T>(keyName(args), args, help, T(), false, false);
+  }
+
+  /**
+   * @brief Switch to different option group or to the default group if
+   * argument is empty
+   *
+   * @param name Header of the option group
+   */
+  void switchGroup(const std::string &name = "");
+
+  // Parse command-line arguments. Handles --help and --version options
+  void parse(int argc, char **argv);
+
+  // Check if an option has been defined (not necessarily parsed)
+  bool has(const std::string &key) const;
+
+  // Get the current value for the option
+  template <typename T>
+  T get(const std::string &key) const {
+    ABORT_IF(
+        vars_.count(key) == 0, "An option with key '{}' is not defined", key);
+    return vars_.at(key)->as<T>();
+  }
+
+  YAML::Node getConfig() const;
```

Comment:
How does this class relate to YAML::Node? Is this function just an accessor, or does it perform an actual deep conversion of a data structure? Say both in a comment.

Reply:
Fixed